go-controller/pkg/ovn/loadbalancer/loadbalancer.go

```diff
@@ -54,34 +54,73 @@ func getLoadBalancerByProtocolType(protocol kapi.Protocol, idkey string) (string
 	return res, nil
 }
 
-// CreateLoadBalancer creates the loadbalancer if it doesn´t exist to avoid
+// Createk8sClusterLoadBalancer creates the k8s-cluster-lb if it doesn´t exist to avoid
 // consumers to create duplicate loadbalancers with the same name and externalID
-func CreateLoadBalancer(protocol kapi.Protocol, idkey string) (string, error) {
+// If it exists it ensures the correct options are set
+func CreateK8sClusterLoadBalancer(protocol kapi.Protocol, idkey string) (string, error) {
+	// check if the reject option should be true for k8's cluster lb
+	reject := true
+	if idkey == types.ClusterIdlingLBPrefix {
+		reject = false
+	}
+	optionsReject := fmt.Sprintf("options:reject=%t", reject)
+
 	lbUUID, err := getLoadBalancerByProtocolType(protocol, idkey)
 	if err != nil && !errors.Is(err, LBNotFound) {
 		return "", errors.Wrapf(err, "Failed to get OVN load balancer for protocol %s", protocol)
 	}
 	// create the load balancer if it doesn't exist yet
 	if lbUUID == "" {
-		lbUUID, err = createLoadBalancer(protocol, idkey)
+		// Create the external ID for the K8's cluster Lbs
+		externalId := fmt.Sprintf("external_ids:%s-%s=yes", idkey, strings.ToLower(string(protocol)))
+		lbUUID, err = createLoadBalancer(protocol, externalId, optionsReject, types.OptionsHairpin)
 		if err != nil {
 			return "", errors.Wrapf(err, "Failed to create OVN load balancer for protocol %s", protocol)
 		}
+	} else {
+		// Generate expected options for LB
+		err = UpdateLoadBalancerOptions(lbUUID, optionsReject, types.OptionsHairpin)
+		if err != nil {
+			return "", errors.Wrapf(err, "Failed to update OVN load balancer for protocol %s", protocol)
+		}
+	}
+
+	return lbUUID, nil
+}
+
+// CreatePerNodeLoadBalancer creates the per node GWR OR Worker loadbalancers
+// (if nodeName is "" we make the GWR Lbs). Unlike CreateK8sClusterLoadBalancer
+// by the time we call this func we are sure the lbs do no exist, so just wrap
```

Comment:
i dont think you should have that assumption here. You are migrating the functionality to a centralized lib, so future callers may not make the same assumption that the current path does. Have the same uuid check here

Reply:
I only did this because of how it was implemented before, all the find operations were batched, I will clean this up and standardize how we interface with OVN LBs (It should all be from the `loadbalancer` package)